onnxruntime/core/graph/graph_viewer.cc

```diff
@@ -262,7 +273,32 @@ bool GraphViewer::IsSubgraph() const {
 }
 
 bool GraphViewer::IsConstantInitializer(const std::string& name, bool check_outer_scope) const {
-  return graph_->GetConstantInitializer(name, check_outer_scope) != nullptr;
+  return GetConstantInitializer(name, check_outer_scope) != nullptr;
+}
+
+const ONNX_NAMESPACE::TensorProto* GraphViewer::GetConstantInitializer(const std::string& initializer_name,
+                                                                       bool check_outer_scope) const {
+  const ONNX_NAMESPACE::TensorProto* initializer = nullptr;
+  if (GetInitializedTensor(initializer_name, initializer)) {
+    if (CanOverrideInitializer()) {
+      const auto& graph_inputs = GetInputsIncludingInitializers();
+      bool is_constant = std::none_of(graph_inputs.cbegin(), graph_inputs.cend(),
+                                      [&initializer_name](const NodeArg* input) {
+                                        return input->Name() == initializer_name;
+                                      });
+
+      if (!is_constant) {
+        initializer = nullptr;
+      }
+    }
+  } else if (check_outer_scope && IsSubgraph()) {
+    // make sure there's not a local value with the same name. if there is it shadows any initializer in outer scope.
+    if (graph_->IsOuterScopeValue(initializer_name)) {
+      initializer = graph_->ParentGraph()->GetConstantInitializer(initializer_name, check_outer_scope);
```

Comment:
>GetConstantInitializer [](http://example.com/codeflow?start=43&length=22)

is it ok to call Graph::GetConstantInitializer() for the parent graph and not ok to do so for the current one?

Reply:
The graph_viewer may have a subset of the initializers of the underlying graph
For the parent graph, since right now the design of the GraphViewer does not really cover this part, ideally a GraphViewer could have a optional parent GraphViewer, (if exists, otherwise use the parent graph)

For now, I think this is fine, since for full ORT, the GraphViewer is actually constructed from a full graph, and for NNAPI, it does not cover subgraph yet,

But we need to log a work item to track this

Reply:
IIRC we filtered the initializers so the list of inputs including initializers was correct. That possibly has different requirements to checking if something is a constant initializer when optimizing, where it may be better to have the complete list. 

That said, is it the case that an initializer lookup would only be coming from a value used by a node in the view, so those values should all be in the filtered initializers, so it would be fine to lookup against the full list? i.e. it's implicitly filtered if you're only looking up values based on the nodes in the view.